src/main/java/com/yelp/nrtsearch/server/luceneserver/QueryNodeMapper.java

```diff
@@ -112,15 +113,11 @@ private PhraseQuery getPhraseQuery(com.yelp.nrtsearch.server.grpc.PhraseQuery ph
 
     private FunctionScoreQuery getFunctionScoreQuery(com.yelp.nrtsearch.server.grpc.FunctionScoreQuery functionScoreQuery,
                                                      IndexState state) {
-        Expression expr;
-        String exprString = functionScoreQuery.getFunction();
-        try {
-            expr = JavascriptCompiler.compile(exprString);
-        } catch (ParseException pe) {
-            // Static error (e.g. bad JavaScript syntax):
-            throw new IllegalArgumentException(String.format("could not parse expression: %s", exprString), pe);
-        }
-        return new FunctionScoreQuery(getQuery(functionScoreQuery.getQuery(), state), expr.getDoubleValuesSource(state.exprBindings));
+        ScoreScript.Factory scriptFactory = ScriptService.getInstance().compile(functionScoreQuery.getScript(), ScoreScript.CONTEXT);
```

Comment:
Nice ! This will allow us to use to binds script parameters into the expression using `getDoubleValuesSource(bindings)`.

JavascriptCompiler.compile(exprString) is a hotspot right now for us